File: exam-system-java/entity/src/main/java/com/yyjy/exam/entity/question/io/res/QuestionGenerateDto.java
package com.yyjy.exam.entity.question.io.res;

import com.yyjy.exam.entity.question.entity.QuestionDifficulty;
import com.yyjy.exam.entity.question.entity.QuestionType;

import java.util.List;

public record QuestionGenerateDto(
		String title,
		QuestionType type,
		QuestionDifficulty difficulty,
		int score,
		String analysis,
		List<String> choices,
		String answer
) {
}

File: exam-system-server/entity/src/main/kotlin/com/yyjy/models/dto/QuestionGenerateDto.kt
package com.yyjy.models.dto

/**
 * @author Nyxcirea
 * @date 2026/4/13
 */
data class QuestionGenerateDto(
    val title: String,
    val type: String,
    val multi: Boolean,
    val difficulty: String,
    val score: Int,
    val analysis: String?,
    val choices: List<String>?,
    val answer: String
)
